frontend/src/metabase/querying/filters/components/FilterValuePicker/ListValuePicker/utils.ts

```diff
@@ -9,3 +12,32 @@ export function searchOptions(
     ({ label }) => label != null && label.toLowerCase().includes(searchValue),
   );
 }
+
+function getSelectedOptions(selectedValues: string[]) {
+  return selectedValues.map((value) => ({
+    value,
+  }));
+}
+
+export function getEffectiveOptions(
+  fieldValues: FieldValue[],
+  selectedValues: string[],
+  elevatedValues: string[] = [],
+): ComboboxItem[] {
+  const options: { label?: string; value: string }[] = [
+    ...getSelectedOptions(elevatedValues),
+    ...getFieldOptions(fieldValues),
+    ...getSelectedOptions(selectedValues),
+  ];
+
+  const mapping = options.reduce((map: Map<string, string>, option) => {
```

Comment:
I would find a simple loop or `forEach` more readable here since we're not actually reducing on the map, but just mutating it.